Sources/WormholySwift/Extensions/UIViewController+Ext.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  Wormholy.swift
+//  Wormholy
+//
+//  Created by Paolo Musolino.
+//  Copyright © 2018 Wormholy. All rights reserved.
+//
+
+import UIKit
+import SwiftUI
+
+extension UIViewController {
+
+    static func currentViewController(_ viewController: UIViewController? = UIApplication.shared.windows.filter(\.isKeyWindow).first?.rootViewController) -> UIViewController? {
+        guard let viewController else { return nil }
+
+        if let viewController = viewController as? UINavigationController {
+            if let viewController = viewController.visibleViewController {
+                return currentViewController(viewController)
+            } else {
+                return currentViewController(viewController.topViewController)
+            }
+        } else if let viewController = viewController as? UITabBarController {
+            if let viewControllers = viewController.viewControllers, viewControllers.count > 5, viewController.selectedIndex >= 4 {
+                return currentViewController(viewController.moreNavigationController)
+            } else {
+                return currentViewController(viewController.selectedViewController)
+            }
```

Comment:
It would be helpful to clarify this logic and determine if it makes sense to add a comment for future reference.